### PR TITLE
External secrets: add readme

### PR DESCRIPTION
**Pull request by Halamix2**
**Description**

Changes proposed in this pull request:

- add readme for external secrets syncer

**Related issue(s)**

#3930

### CODE REVIEW

**Review by mmitoraj — COMMENTED** [review]
Please see my comments below. Thank you.

File: prow/cluster/resources/external-secrets/README.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+Kubernetes secrets are synchronized with GCP Secret Manager using [Kubernetes External Secrets](https://github.com/external-secrets/kubernetes-external-secrets).
+
+# Installation
+
+Follow these steps to install `kubernetes-external-secrets` on an untrusted cluster in the `external-secrets` namespace:
+
+1. Create the `external-secrets` namespace. Run `kubectl create namespace external-secrets`.
+2. Add the `sa-secret-manager-untrusted` secret containing credentials for GCP service account with permission to access secrets.
+3. Add `external-secrets` Helm repo. Run `helm repo add external-secrets https://external-secrets.github.io/kubernetes-external-secrets/`.
+4. Install `external-secrets/kubernetes-external-secrets` Helm chart. Run `helm install -f prow/cluster/resources/external-secrets/values_untrusted.yaml -n external-secrets kubernetes-external-secrets external-secrets/kubernetes-external-secrets`.
```
Comment (mmitoraj) [inline prow/cluster/resources/external-secrets/README.md:14]:
```suggestion
4. Install  the`external-secrets/kubernetes-external-secrets` Helm chart. Run `helm install -f prow/cluster/resources/external-secrets/values_untrusted.yaml -n external-secrets kubernetes-external-secrets external-secrets/kubernetes-external-secrets`.
```

File: prow/cluster/resources/external-secrets/README.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+# Configuration
```
Comment (mmitoraj) [inline prow/cluster/resources/external-secrets/README.md:15]:
```suggestion

# Configuration
```

Comment (mmitoraj) [reply prow/cluster/resources/external-secrets/README.md:15]: Please add an empty line above the header.

**Review by mmitoraj — COMMENTED** [review]
Have a look at my suggestions below. Thank you.

**Review by mmitoraj — APPROVED** [review]

File: prow/cluster/resources/external-secrets/README.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
+# External secrets
```
Comment (mmitoraj) [inline prow/cluster/resources/external-secrets/README.md:1]:
```suggestion
# External Secrets
```

File: prow/cluster/resources/external-secrets/README.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+## Overview
+
+Kubernetes secrets are synchronized with GCP Secret Manager using [Kubernetes External Secrets](https://github.com/external-secrets/kubernetes-external-secrets).
```
Comment (mmitoraj) [inline prow/cluster/resources/external-secrets/README.md:5]:
```suggestion
Kubernetes Secrets are synchronized with GCP Secret Manager using [Kubernetes External Secrets](https://github.com/external-secrets/kubernetes-external-secrets).
```

File: prow/cluster/resources/external-secrets/README.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+# Installation
+
+Follow these steps to install `kubernetes-external-secrets` on an untrusted cluster in the `external-secrets` Namespace:
+
+1. Create the `external-secrets` Namespace. Run `kubectl create namespace external-secrets`.
+2. Add the `sa-secret-manager-untrusted` secret containing credentials for a GCP service account with permission to access secrets.
```
Comment (mmitoraj) [inline prow/cluster/resources/external-secrets/README.md:12]:
```suggestion
2. Add the `sa-secret-manager-untrusted` Secret containing credentials for a GCP service account with permission to access Secrets.
```

File: prow/cluster/resources/external-secrets/README.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+3. Add the `external-secrets` Helm repository. Use the following command: `helm repo add external-secrets https://external-secrets.github.io/kubernetes-external-secrets/`.
+4. Install the `external-secrets/kubernetes-external-secrets` Helm chart. Run `helm install -f prow/cluster/resources/external-secrets/values_untrusted.yaml -n external-secrets kubernetes-external-secrets external-secrets/kubernetes-external-secrets`.
+
+# Configuration
+
+Secrets can be stored as text in GCP Secret Manager and be mapped to a Kubernetes secret with one key.
```
Comment (mmitoraj) [inline prow/cluster/resources/external-secrets/README.md:18]:
```suggestion
Secrets can be stored as text in GCP Secret Manager and be mapped to a Kubernetes Secret with one key. 
```

File: prow/cluster/resources/external-secrets/README.md [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+spec:
+  backendType: gcpSecretsManager
+  projectId: my-gcp-project
+  data:
+    - key: gcp-plain-secret # name of the GCP secret
+      name: token # key name in the k8s secret
+      version: latest # version of the GCP secret
+```
+
+Secrets can also be stored as JSON in GCP Secret Manager and be mapped to a Kubernetes secret with multiple keys.
```
Comment (mmitoraj) [inline prow/cluster/resources/external-secrets/README.md:36]:
```suggestion
Secrets can also be stored as JSON in GCP Secret Manager and be mapped to a Kubernetes Secret with multiple keys. 
```

File: prow/cluster/resources/external-secrets/README.md [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+    - key: gcp-json-secret # name of the GCP secret
+      name: anotherKey # key name in the k8s secret
+      version: latest # version of the GCP secret
+      property: anotherKey # name of the field in the GCP secret JSON, unused for plain values
+```
+
+
+# External Secrets Checker
+
+External Secrets Checker checks if all External Secrets have synchronized succesfully, and if all Secrets have corresponding External Secrets.
```
Comment (mmitoraj) [inline prow/cluster/resources/external-secrets/README.md:62]:
```suggestion
External Secrets Checker checks if all External Secrets synchronized successfully, and if all Secrets have corresponding External Secrets.
```

File: prow/cluster/resources/external-secrets/README.md [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+
+Secrets can also be stored as JSON in GCP Secret Manager and be mapped to a Kubernetes secret with multiple keys. 
+
+See an example:
+
+```yaml
+apiVersion: kubernetes-client.io/v1
+kind: ExternalSecret
+metadata:
+  name: secretName # name of the k8s external secret and the k8s secret
```
Comment (mmitoraj) [inline prow/cluster/resources/external-secrets/README.md:44]:
```suggestion
  name: secretName # name of the k8s external Secret and the k8s Secret
```

File: prow/cluster/resources/external-secrets/README.md [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+spec:
+  backendType: gcpSecretsManager
+  projectId: my-gcp-project
+  data:
+    - key: gcp-json-secret # name of the GCP secret
```
Comment (mmitoraj) [inline prow/cluster/resources/external-secrets/README.md:49]:
```suggestion
    - key: gcp-json-secret # name of the GCP Secret
```

File: prow/cluster/resources/external-secrets/README.md [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+      name: keyName # key name in the k8s secret
```
Comment (mmitoraj) [inline prow/cluster/resources/external-secrets/README.md:50]:
```suggestion
      name: keyName # key name in the k8s Secret
```

File: prow/cluster/resources/external-secrets/README.md [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+      version: latest # version of the GCP secret
```
Comment (mmitoraj) [inline prow/cluster/resources/external-secrets/README.md:51]:
```suggestion
      version: latest # version of the GCP Secret
```

File: prow/cluster/resources/external-secrets/README.md [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+      property: keyName # name of the field in the GCP secret JSON, unused for plain values
```
Comment (mmitoraj) [inline prow/cluster/resources/external-secrets/README.md:52]:
```suggestion
      property: keyName # name of the field in the GCP Secret JSON, unused for plain values
```

File: prow/cluster/resources/external-secrets/README.md [hunk 14]
```diff
@@ -0,0 +1,64 @@
...
+    - key: gcp-json-secret # name of the GCP secret
```
Comment (mmitoraj) [inline prow/cluster/resources/external-secrets/README.md:53]:
```suggestion
    - key: gcp-json-secret # name of the GCP Secret
```

File: prow/cluster/resources/external-secrets/README.md [hunk 15]
```diff
@@ -0,0 +1,64 @@
...
+      name: anotherKey # key name in the k8s secret
```
Comment (mmitoraj) [inline prow/cluster/resources/external-secrets/README.md:54]:
```suggestion
      name: anotherKey # key name in the k8s Secret
```

File: prow/cluster/resources/external-secrets/README.md [hunk 16]
```diff
@@ -0,0 +1,64 @@
...
+      version: latest # version of the GCP secret
```
Comment (mmitoraj) [inline prow/cluster/resources/external-secrets/README.md:55]:
```suggestion
      version: latest # version of the GCP Secret
```

File: prow/cluster/resources/external-secrets/README.md [hunk 17]
```diff
@@ -0,0 +1,64 @@
...
+      property: anotherKey # name of the field in the GCP secret JSON, unused for plain values
```
Comment (mmitoraj) [inline prow/cluster/resources/external-secrets/README.md:56]:
```suggestion
      property: anotherKey # name of the field in the GCP Secret JSON, unused for plain values
```

**Review by mmitoraj — CHANGES_REQUESTED** [review]
Hi, I left my comments, please capitalize the word `Secrets` and correct the last section. Thank you.

File: prow/cluster/resources/external-secrets/README.md [hunk 18]
```diff
@@ -0,0 +1,64 @@
...
+      version: latest # version of the GCP Secret
+      property: anotherKey # name of the field in the GCP Secret JSON, unused for plain values
+```
+
+
+# External Secrets Checker
+
+External Secrets Checker checks if all External Secrets synchronized successfully, and if all Secrets have corresponding External Secrets.
+
+It can be installed with the `kubectl apply -f external_secrets_checker_prow.yaml` command.
```
Comment (mmitoraj) [inline prow/cluster/resources/external-secrets/README.md:64]:
```suggestion
To install External Secrets Checker run the following command: 

`kubectl apply -f external_secrets_checker_prow.yaml`
```

**Review by mmitoraj — APPROVED** [review]

**Review by adamwalach — APPROVED** [review]

**Pull request merged**